petab/simulate.py

```diff
@@ -200,9 +221,20 @@ def sample_noise(
         .loc[measurement_row[petab.C.OBSERVABLE_ID]]
         .get(petab.C.NOISE_DISTRIBUTION, petab.C.NORMAL)
     )
+    # an empty noise distribution column in an observables table can result in
+    # `noise_distribution == float('nan')`
+    if pd.isna(noise_distribution):
+        noise_distribution = petab.C.NORMAL
 
     # below is e.g.: `np.random.normal(loc=simulation, scale=noise_value)`
-    return getattr(rng, noise_distribution)(
+    simulated_value_with_noise = getattr(rng, noise_distribution)(
         loc=simulated_value,
         scale=noise_value * noise_scaling_factor
     )
+
+    if (
+            zero_bounded and
+            np.sign(simulated_value) != np.sign(simulated_value_with_noise)
+    ):
+        return 0
```

Comment:
```suggestion
        return 0.0
```

Since the function promises to return a `float`